### PR TITLE
feat(core-components): Introduced useSignInProviderElements hook to support building custom login pages

### PR DESCRIPTION
**Pull request by jboeijenga**
## Hey, I just made a Pull Request!

Introduced new useSignInProviderElements hook to solve: https://github.com/backstage/backstage/issues/26423

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by camilaibs — APPROVED** [review]
It sounds like a good approach, thanks for contributing with us 🙂

File: packages/core-components/src/layout/SignInPage/providers.tsx
Comment (jboeijenga) [reply packages/core-components/src/layout/SignInPage/providers.tsx]: Thanks for the suggestion!

**Comment by jboeijenga** [comment]
@camilaibs: thanks for your approval, what is needed to get this merged?

**Review by mounicarajput — APPROVED** [review]

**Comment by camilaibs** [comment]
> @camilaibs: thanks for your approval, what is needed to get this merged?

We need one more @backstage/maintainers approval 🙂

**Comment by jboeijenga** [comment]
@benjdlambert  Can you please review this change?

File: packages/core-components/api-report.md
Comment (benjdlambert) [inline packages/core-components/api-report.md:1518]: I think that we probably want to change the returned API a little bit here from the `useSignInProviders`.

It looks like it returns an array of `loading` and `elements`.

https://github.com/backstage/backstage/blob/8743beff4d843cc0fe0896a04407369f2db5c902/packages/core-components/src/layout/SignInPage/providers.tsx#L203

I think that we should probably make that API a little bit friendlier. Maybe

`return { loading, elements }` or something instead?

File: packages/core-components/src/layout/SignInPage/providers.tsx
Comment (Rugvip) [inline packages/core-components/src/layout/SignInPage/providers.tsx:228]: I'm a bit torn on this API, there's a lot baked into the hook, with the card + `Grid item` wrapping. Would a slightly higher level API work for you? I'm thinking a potentially `SignInProviderItems` that more or less mirrors the existing `SignInPage` but without the page wrapping, probably with the `Grid container` as the root element?

**Pull request closed without merging**
